sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/core/compiler/model/NamedQuery.kt

```diff
@@ -104,7 +105,15 @@ data class NamedQuery(
     pureTable?.let {
       return@lazy ClassName(it.tableName.sqFile().packageName!!, allocateName(it.tableName).capitalize())
     }
-    return@lazy ClassName(select.sqFile().packageName!!, name.capitalize())
+    var packageName = select.sqFile().packageName!!
+    if (select.sqFile().parent?.files
+      ?.filterIsInstance<SqlDelightQueriesFile>()?.flatMap { it.namedQueries }
+      ?.filter { it.needsInterface() && it != this }
+      ?.any { it.name == name } == true
+    ) {
+      packageName = "$packageName.${select.sqFile().virtualFile!!.nameWithoutExtension.decapitalize()}"
```

Comment:
```suggestion
      packageName = "$packageName.${select.sqFile().virtualFile!!.nameWithoutExtension.decapitalize(Locale.ROOT)}"
```
Or use US.

Currently you might produce different package names based on system locale (specifically, Turkish).